tests/integration/mirrorpeersecret_controller_test.go

```diff
@@ -0,0 +1,534 @@
+		{
+			ClusterName: "cluster1",
+			StorageClusterRef: multiclusterv1alpha1.StorageClusterRef{
+				Name:      "test-storagecluster-1",
+				Namespace: "test-namespace",
+			},
+		},
+		{
+			ClusterName: "cluster2",
+			StorageClusterRef: multiclusterv1alpha1.StorageClusterRef{
+				Name:      "test-storagecluster-2",
+				Namespace: "test-namespace",
+			},
+		},
+	}
+	return createMirrorPeer(defaultMirrorPeerName, items...)
+}
+
+func createFakeScheme() *runtime.Scheme {
+	// groupVersion is group version used to register these objects
+	groupVersion := rtscheme.GroupVersion{Group: "orchestrator.odf.openshift.io", Version: "v1"}
+	// schemeBuilder is used to add go types to the GroupVersionKind scheme
+	schemeBuilder := &pkgscheme.Builder{GroupVersion: groupVersion}
+
+	scheme, err := schemeBuilder.Build()
+	utilruntime.Must(err)
+
+	utilruntime.Must(schemeBuilder.AddToScheme(scheme))
+	utilruntime.Must(clientgoscheme.AddToScheme(scheme))
+	utilruntime.Must(clusterv1.AddToScheme(scheme))
+	utilruntime.Must(multiclusterv1alpha1.AddToScheme(scheme))
+	utilruntime.Must(addonapiv1alpha1.AddToScheme(scheme))
+
+	return scheme
+}
+
+func createFakeClient() client.Client {
+	scheme := createFakeScheme()
+	fc := fake.NewClientBuilder().
+		WithScheme(scheme).Build()
+	return fc
+}
+
+var _ = Describe("MirrorPeerSecret Controller", func() {
+	var (
+		ctx           context.Context
+		mirrorPeerPP  MirrorPeerPlusPlus
+		k8sClient     client.Client
+		err           error
+		sourceSecrets []corev1.Secret
+	)
+
+	BeforeEach(func() {
+		k8sClient = createFakeClient()
+		Expect(err).To(BeNil())
+		mirrorPeerPP = MirrorPeerPlusPlus(createDefaultMirrorPeer())
+		ctx = context.TODO()
+
+		By("creating managed clusters", func() {
+			managedClusters := mirrorPeerPP.GetManagedClusters()
+			for _, eachMC := range managedClusters {
+				err := k8sClient.Create(ctx, &eachMC)
+				Expect(err).To(BeNil())
+			}
+		})
+		By("creating the namespaces", func() {
+			namespaces := mirrorPeerPP.GetClusterNamespaces()
+			for _, eachNspace := range namespaces {
+				err := k8sClient.Create(ctx, &eachNspace)
+				Expect(err).To(BeNil())
+			}
+		})
+		By("creating the MirrorPeer", func() {
+			mirrorPeer := mirrorPeerPP.GetMirrorPeer()
+			err := k8sClient.Create(ctx, &mirrorPeer)
+			Expect(err).To(BeNil())
+		})
+		By("creating the Source secrets", func() {
+			sourceSecrets = mirrorPeerPP.GetDummySourceSecrets("my-source-secret")
+			for _, sourceSecret := range sourceSecrets {
+				err := k8sClient.Create(ctx, &sourceSecret)
+				Expect(err).To(BeNil())
+			}
+		})
+	})
+
+	JustBeforeEach(func() {
+		By("doing a Reconcile on MirrorPeer controller", func() {
+			var mp = controllers.MirrorPeerReconciler{
+				Client: k8sClient,
+				Scheme: k8sClient.Scheme(),
+			}
+			mirrorPeerNN := types.NamespacedName{Name: mirrorPeerPP.Name}
+			_, err = mp.Reconcile(ctx, ctrl.Request{NamespacedName: mirrorPeerNN})
+			Expect(err).To(BeNil())
+		})
+	})
```

Comment:
A reconciler is already running when the test starts. No need to call it explicitly.

Reply:
Without this again tests are failing...